nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -175,6 +181,32 @@ public QueryDatabaseTable() {
         return propDescriptors;
     }
 
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>(super.customValidate(validationContext));
+
+        final String tableName = validationContext.getProperty(TABLE_NAME).getValue();
+        final String sqlQuery = validationContext.getProperty(SQL_QUERY).getValue();
+
+        if(!StringUtils.isEmpty(sqlQuery) && !StringUtils.isEmpty(tableName)){
+            results.add(new ValidationResult.Builder()
+                        .valid(false)
+                        .subject("SQL Query")
+                        .explanation("SQL Query and Table Name can't both be specified at the same time.")
+                        .build());
+        }
+
+        if(!StringUtils.isEmpty(sqlQuery) && isDynamicMaxValues){
```

Comment:
Is this a hard requirement? Expression Language support (since incoming connections are not allowed for QueryDatabaseTable) is meant to support Variable Registry and environment variables, which could be used in a custom query if it also included such an EL statement?